contracts/Strategy.sol

```diff
@@ -39,66 +24,119 @@ interface IUni{
  *
  ********************* */
 
-contract Strategy is BaseStrategy, DydxFlashloanBase, ICallee {
+contract Strategy is BaseStrategy, ICallee {
     using SafeERC20 for IERC20;
     using Address for address;
     using SafeMath for uint256;
 
     // @notice emitted when trying to do Flash Loan. flashLoan address is 0x00 when no flash loan used
     event Leverage(uint256 amountRequested, uint256 amountGiven, bool deficit, address flashLoan);
 
-    //Flash Loan Providers
-    address private constant SOLO = 0x1E0447b19BB6EcFdAe1e4AE1694b0C3659614e4e;
-
     // Comptroller address for compound.finance
     ComptrollerI private constant compound = ComptrollerI(0x3d9819210A31b4961b30EF54bE2aeD79B9c9Cd3B);
 
     //Only three tokens we use
     address private constant comp = 0xc00e94Cb662C3520282E6f5717214004A7f26888;
     CErc20I public cToken;
-    //address public constant DAI = address(0x6B175474E89094C44Da98b954EedeAC495271d0F);
 
-    address public constant uniswapRouter = 0x7a250d5630B4cF539739dF2C5dAcb4c659F2488D;
+
+    bool public useUniV3;
+    IUniswapV2Router02 public currentV2Router;
+    IUniswapV2Router02 private constant UNI_V2_ROUTER =
+        IUniswapV2Router02(0x7a250d5630B4cF539739dF2C5dAcb4c659F2488D);
+    IUniswapV2Router02 private constant SUSHI_V2_ROUTER =
+        IUniswapV2Router02(0xd9e1cE17f2641f24aE83637ab66a2cca9C378B9F);
+    IUniswapV3Router private constant UNI_V3_ROUTER =
+        IUniswapV3Router(0xE592427A0AEce92De3Edee1F18E0157C05861564);
     address private constant weth = 0xC02aaA39b223FE8D0A0e5C4F27eAD9083C756Cc2;
 
+    uint24 public compToWethSwapFee = 3000;
+    uint24 public wethToWantSwapFee = 3000;
+
     //Operating variables
-    uint256 public collateralTarget = 0.73 ether; // 73%
+    uint256 public collateralTarget = 0.63 ether; // 63%
     uint256 public blocksToLiquidationDangerZone = 46500; // 7 days =  60*60*24*7/13
 
     uint256 public minWant; // Default is 0. Only lend if we have enough want to be worth it. Can be set to non-zero
     uint256 public minCompToSell = 0.1 ether; //used both as the threshold to sell but also as a trigger for harvest
+    bool public dontClaimComp;
 
     //To deactivate flash loan provider if needed
     bool public DyDxActive = true;
-
     bool public forceMigrate; // default is false
 
-    uint256 public dyDxMarketId;
+    constructor(address _vault, address _cToken) public BaseStrategy(_vault) {
+        _initializeThis(_cToken);
+    }
 
+    function approveTokenMax(address token, address spender) internal {
+        IERC20(token).safeApprove(spender, type(uint256).max);
+    }
 
+    // To receive ETH from compound and WETH contract
+    receive() external payable {}
 
-    constructor(address _vault, address _cToken) public BaseStrategy(_vault) {
+    function name() external override view returns (string memory){
+        return "StrategyGenericLevCompFarm";
```

Comment:
i think the change is big enough to add a V2 at the end of this, so we can make a new risk group for this contract and score it separately from original

Reply:
agree, changed to `GenLevCompV2`